src/icrar/leap-accelerate/algorithm/casa/PhaseRotate.cc

```diff
@@ -143,37 +198,40 @@ namespace casalib
         assert(metadata.oldUVW.size() == integration.baselines);
         assert(metadata.channel_wavelength.size() == metadata.channels);
 
+        std::cout << "uvw is " << uvw[0] << std::endl;
+
         // loop over baselines
-        // for(int baseline = 0; baseline < integration.baselines; ++baseline)
-        // {
-        //     // For baseline
-        //     const double pi = boost::math::constants::pi<double>();
-        //     double shiftFactor = -2 * pi * uvw[baseline].get()[2] - metadata.oldUVW[baseline].get()[2]; // check these are correct
-        //     shiftFactor = shiftFactor + 2 * pi * (metadata.phase_centre_ra_rad * metadata.oldUVW[baseline].get()[0]);
-        //     shiftFactor = shiftFactor - 2 * pi * (direction.get()[0] * uvw[baseline].get()[0] - direction.get()[1] * uvw[baseline].get()[1]);
-
-        //     if(baseline % 1000 == 1)
-        //     {
-        //         std::cout << "ShiftFactor for baseline " << baseline << " is " << shiftFactor << std::endl;
-        //     }
-
-        //     // Loop over channels
-        //     for(int channel = 0; channel < metadata.channels; channel++)
-        //     {
-        //         double shiftRad = shiftFactor / metadata.channel_wavelength[channel];
-
-        //         Eigen::VectorXcd v = data(channel, baseline);
-        //         data(channel, baseline) = v * std::exp(std::complex<double>(0.0, 1.0) * std::complex<double>(shiftRad, 0.0));
-
-        //         if(!data(channel, baseline).hasNaN())
-        //         {
-        //             for(int i = 0; i < data(channel, baseline).cols(); i++)
-        //             {
-        //                 metadata.avg_data.get()(baseline, i) += data(channel, baseline)(i);
-        //             }
-        //         }
-        //     }
-        // }
+        for(int baseline = 0; baseline < integration.baselines; ++baseline)
+        {
+            // For baseline
+            const double pi = boost::math::constants::pi<double>();
+            double shiftFactor = -2 * pi * uvw[baseline].get()[2] - metadata.oldUVW[baseline].get()[2]; // check these are correct
+
+            shiftFactor = shiftFactor + 2 * pi * (metadata.phase_centre_ra_rad * metadata.oldUVW[baseline].get()[0]);
+            shiftFactor = shiftFactor - 2 * pi * (direction.get()[0] * uvw[baseline].get()[0] - direction.get()[1] * uvw[baseline].get()[1]);
```

Comment:
Like in #21 `2 * pi` can be factored out and `shiftFactor` can be changed via `+=` an `-=`.

Reply:
I also don't know if it makes sense to reuse the same code in both places actually. But I remember you wanted one of the implementations to be the reference, and hence code reuse didn't make sense in that scenario.

Reply:
This particular file and the icrar::casalib namespace is just a direct port of the python implementation and still needs to be verified as producing the same identical output. Once the other implementations have been verified as working this can be removed as it's not part of the software requirements